CI: ERROR at setup of TestCasting.test_astype_string with pandas master

We have a failure with pandas master caused by https://github.com/pandas-dev/pandas/pull/40998. See the [CI log](https://github.com/geopandas/geopandas/runs/2484443872?check_suite_focus=true#step:6:2962).

It is one of the pandas extension tests we inherit and I am not sure if it should be fixed/skipped on our side or in pandas.

cc @jorisvandenbossche 

```py
_______________ ERROR at setup of TestCasting.test_astype_string _______________
[gw0] linux -- Python 3.8.8 /usr/share/miniconda/envs/test/bin/python
file /usr/share/miniconda/envs/test/lib/python3.8/site-packages/pandas/tests/extension/base/casting.py, line 46
      def test_astype_string(self, data, nullable_string_dtype):
E       fixture 'nullable_string_dtype' not found
>       available fixtures: add_geopandas, all_arithmetic_operators, all_boolean_reductions, all_compare_operators, all_data, all_numeric_reductions, as_array, as_frame, as_series, box_in_series, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, data, data_for_grouping, data_for_sorting, data_for_twos, data_missing, data_missing_for_sorting, data_repeated, doctest_namespace, dtype, fillna_method, groupby_apply_op, monkeypatch, na_cmp, na_value, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, use_numpy, worker_id
```

--- a/geopandas/array.py
+++ b/geopandas/array.py
@@ -1030,7 +1030,7 @@ class GeometryArray(ExtensionArray):
                 pd_dtype = pd.api.types.pandas_dtype(dtype)
                 if isinstance(pd_dtype, pd.StringDtype):
                     # ensure to return a pandas string array instead of numpy array
-                    return pd.array(string_values, dtype="string")
+                    return pd.array(string_values, dtype=pd_dtype)
             return string_values.astype(dtype, copy=False)
         else:
             return np.array(self, dtype=dtype, copy=copy)

